src/components/TodoItem.tsx

```diff
@@ -0,0 +1,51 @@
+import { useState } from "react";
+
+interface TodoItemProps {
+  todo: {
+    title: string;
+    date: string;
+    complete: boolean;
+  };
```

Comment:
We can use the `Todo` interface declared in the `TodoList` component here.

```suggestion
todo: Todo
```